plugins/osdn/vnids.go

```diff
@@ -18,14 +20,104 @@ const (
 	AdminVNID = uint(0)
 )
 
+func (oc *OsdnController) GetVNID(name string, retry bool) (uint, error) {
+	oc.vnidLock.Lock()
+	defer oc.vnidLock.Unlock()
+
+	if id, ok := oc.vnidMap[name]; ok {
+		return id, nil
+	}
+
+	// Nodes asynchronously watches for both NetNamespace and service
+	// NetNamespaces populates vnid map and services/pod-setup depend on vnid map
+	// If for some reason, vnid map propagation from master to node is slow
+	// and if service/pod-setup tries to lookup vnid map then it may fail.
+	// So, this retry option will alleviate this problem.
+	if retry {
+		// Try few times up to 2 seconds
+		retries := 20
+		retryInterval := 100 * time.Millisecond
+		for i := 0; i < retries; i++ {
+			oc.vnidLock.Unlock()
+			time.Sleep(retryInterval)
```

Comment:
Will this conflict with someone else?  Should the wait vary by the number or retries or something so that two don't wake up in sync and then sleep the same length?